docs/how-to-guides/develop-a-tool/use-file-path-as-tool-input.md

```diff
@@ -0,0 +1,98 @@
+# Use FilePath as tool input
+
+In this document, we will guide you through the detailed process of how to use FilePath as tool input with 'hello-world' tool example.
+
+## Prerequisites
+
+Since FilePath is still in test currently, you need to install private version SDK and Extension.
+
+1. Download private version promptflow SDK
+
+    ```bash
+    pip install "promptflow==0.0.106113823" --extra-index-url https://azuremlsdktestpypi.azureedge.net/test-promptflow/
+    ```
+
+2. Download this [VSCode Extension](https://aka.ms/pfvsctest), install from VSIX:
+
+    ![install from vsix](../../media/how-to-guides/develop-a-tool/install_from_vsix.png)
+
+## Use FilePath as tool input in 'hello-world' tool
+
+For details of how to create and use custom tool, please refer to [Custom tool package creation and usage](../how-to-create-and-use-your-own-tool-package.md).
+
+1. Create a tool project hello-world-proj with `--case file_input` to generate demo case of file_path as tool input.
+
+    ```bash
+    python <path-to-scripts>\tool\generate_tool_package_template.py --case file_path --destination hello-world-proj --package-name hello-world --tool-name hello_world_tool --function-name get_sqrt    
+    ```
+
+2. Go to hello-world-proj folder, check your tool code.
+
+   The demo use a sematic_kernel case, refer to [run native code with Semantic Kernel](https://learn.microsoft.com/en-us/semantic-kernel/ai-orchestration/plugins/native-functions/using-the-skfunction-decorator?tabs=python).
+
+   ![update custom tool](../../media/how-to-guides/develop-a-tool/generate_custom_tool_with_file_path.png)
+
+3. Build and install tool package in VSCode Extension, according to [Custom tool package creation and usage](../how-to-create-and-use-your-own-tool-package.md).
+
+   Build tool:
+
+   ```bash
+   python setup.py sdist bdist_wheel
+   ```
+
+   Install tool in vscode:
+
+   ```bash
+   pip install dist\hello_world-0.0.1-py3-none-any.whl
+   ```
+
+4. Go to the extension and create new flow. Click 'empty flow' and preview the flow. Next, click `+ More` button and select your tool `Hello World Tool`
+
+   ![create empty flow](../../media/how-to-guides/develop-a-tool/create_empty_flow.png)
+   ![use hello world tool](../../media/how-to-guides/develop-a-tool/select_hello_world_tool.png)
+
+5. Prepare your own python script, `demo_input.py` as file_input.
+
+   Here the input demos a custom sqrt func, for more details, refer to [Use the SKFunction decorator to define a native function](https://learn.microsoft.com/en-us/semantic-kernel/ai-orchestration/plugins/native-functions/using-the-skfunction-decorator?tabs=python#use-the-skfunction-decorator-to-define-a-native-function)
+
+   ```python
+    import math
+    
+    from semantic_kernel.skill_definition import sk_function, sk_function_context_parameter
```

Comment:
this example tool requires `semantic-kernel` package, where do you add the dependency? Maybe we can add a simple example.